src/libraries/System.Private.Xml.Linq/tests/TrimmingTests/XElementCtor.cs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.Serialization;
+
+public class Program
+{
+	public static int Main()
+	{
+		var serializer = new MyClassSerializer();
+		var xmlReader = CreateXmlReader();
+
+		var obj = (MyClass)serializer.Deserialize(xmlReader);
```

Comment:
Can we add a comment above this call to explain where in the Deserialize call do we expect some dynamic dependency to get loaded? That way is easier to understand what are we trying to guard against. Also, why do we need to check if obj.Element is null in the next line, if the dynamic dependency is broken I assume that the Deserialize method should throw wouldn't it?

Reply:
I've added more comments.

Yes, the Deserialize call would fail with a MissingMethodException. I added the null check to make sure the end-to-end routine works as expected, which helps validate that the sgen output was copied correctly. I've updated the code to assert on the content of the property, which is maybe clearer than just a null check.